Paper: Text and data mining scientific articles with allofplos

eseiver opened the pull request:
Submitting a paper to SciPy proceedings on the allofplos Python package.

eseiver committed:
add .rst of original proposal

sbassi committed:
Adding DB related text

Added some items related DB and ORM. Also the docker site.

eseiver committed:
WIP: first draft of ms

eseiver committed:
paper updates

mpacer committed:
formatting improvements

mpacer committed:
quote fixing

eseiver committed:
add xpath

mpacer committed:
update xpath formatting

eseiver committed:
add intro

thx @mpacer

githomin commented on papers/elizabeth_seiver/seivpacbassi.rst:
```diff
@@ -0,0 +1,233 @@
+    >>> acknowledge = art.tree.xpath('//ack/p')[0]
+    >>> acknowledge.text
+    'We thank all contributors to the Performance Curve Database (pcdb.santafe.edu).'
+  
```
I think we could use a new header here.

githomin commented on papers/elizabeth_seiver/seivpacbassi.rst:
```diff
@@ -0,0 +1,233 @@
+            else:
+                pass
+
+Query with peewee & SQLite
```
I'm not quite sure what this section means.

eseiver replied:
@sbassi will you have time to write the peewee section?

githomin reviewed (commented):
Looks great. The ending is a bit abrupt. Also, make sure to take care of line-breaks. A lot of the code examples run off the page in the pdf as the layout is 2 columns.

ankurankan commented:
The paper looks great and allofplos seems like a very useful tool. I do have a few comments for possible improvements:
1. As mentioned by @githomin also, the ending seems a bit abrupt. I think adding a conclusion section would help a lot.
2. After the introduction section, I would have liked to know about any similar previous work if any. 
3. The titles of the sections are a bit too informal in my opinion.
4. All the information has been presented really well. But I think the paper could use some work on connecting these sections with each other. Adding leading sentences in the sections or ending sections with what we expect in the next section would really help.
5. Some restructuring (or adding new sections) could make the content much easier to go through. For example, in the section "How allofplos maintains corpora", you have talked about how to download articles which I think is a use case. Therefore, separating these into different sections will make it more structured.

demba commented:
Hi,

I have been asked to provide comments on your paper for the proceedings.

Overall, this is a nice contribution and I commend the authors for putting this tool together.

While all the necessary content is there, my feeling is that

1. The narrative is disconnected: this can be improved by adding trailing sentences after each section hinting at what will be talked about in the next section.

2. The writing feels a bit colloquial for a conference proceeding: the authors use "It's geared" as opposed to "It is geared", or "the article data won't" as opposed to "the article data will not" or "If you've used" vs "If you used".  

3. As mentioned by others, the ending is abrupt indeed.

Looking forward to reading a more polished version.

reddyts commented:
Here are some comments regarding your paper:

" It also quickly and easily parses these article files into Python data structures."
I would recommend you rewrite this sentence as:
"It parses these article files into Python data structures efficiently." (not sure that there is any value in "quickly and easily").

"This article will cover how allofplos keeps your articles up-to-date, and how to use it to easily access common article metadata and fuel your meta-research, with actual use cases from inside PLOS." 
I would recommend you break this into two sentenses.

Reference and/or exlapantion of JATS format. (Please assume that the audience for SciPy does not know about JATS format).
Reference And/Or explanation of DOI. (Please assume that the audience for SciPy does not know about DOI, hows its created/computed, and its importance).

As mentioned by the other reviewers, the ending is not appropriate. The paper needs more organization. To benifit the Scipy audience, it would be better to add more details to your paper. Specifically, by adding  some examples of how people have used allofplos.

There needs to some mention of related work. Do similar systems exist? Can this be expanded to beyond plos? A brief intro plos would be more helpful. Details such as what journals are covered under plos and/or what fields do these journals cover?

dalippa commented:
@eseiver There has been no activity on the paper for the past 11 days.

@reddyts @demba @ankurankan @githomin is this paper ready to be accepted?

reddyts commented:
I do not think this paper is ready for publishing. None of the comments
seem to have been incorporated yet.

On Fri, Jun 29, 2018 at 12:35 PM David Lippa <notifications@github.com>
wrote:

> @eseiver <https://github.com/eseiver> There has been no activity on the
> paper for the past 11 days.
>
> @reddyts <https://github.com/reddyts> @demba <https://github.com/demba>
> @ankurankan <https://github.com/ankurankan> @githomin
> <https://github.com/githomin> is this paper ready to be accepted?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipy-conference/scipy_proceedings/pull/417#issuecomment-401407733>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AmSnD4VLcO25bSq_o2fDmox9oNdcfBp1ks5uBlc3gaJpZM4ULpaW>
> .
>


-- 
Regards,
T. S. Reddy

ankurankan commented:
I don't think this paper is ready either.

eseiver commented:
I agree, sorry have not had time to make these changes. literally in the middle of moving

eseiver committed:
fixing language

eseiver committed:
adding cites

add cites

add DOI cite

eseiver committed:
rename headers

eseiver committed:
+ future directions

sbassi committed:
Update sqlite info

eseiver commented:
Have addressed many of these comments and will continue with more edits

eseiver committed:
adding transitions

eseiver committed:
Merge pull request #6 from eseiver/sqlite

Update sqlite info

mpacer committed:
reformats code to not run over other sections where it can be avoided

eseiver committed:
Merge pull request #7 from mpacer/allofplos_paper

reformats code to not run over other sections where it can be avoided

mpacer committed:
add conclusions section

eseiver commented:
Based on the reviews, we made a number of changes to the paper. Here are the hashes of the commits that address each point with additional explanations where necessary.

> Looks great. The ending is a bit abrupt. Also, make sure to take care of line-breaks. A lot of the code examples run off the page in the pdf as the layout is 2 columns.

Line breaks fixed in f543f7d. Conclusion in a4fa65f

> As mentioned by @githomin also, the ending seems a bit abrupt. I think adding a conclusion section would help a lot.

Conclusion in a4fa65f

> After the introduction section, I would have liked to know about any similar previous work if any.

eLife in c5be880, PMC OA subset in 751df50

> The titles of the sections are a bit too informal in my opinion.

323a50c

> All the information has been presented really well. But I think the paper could use some work on connecting these sections with each other. Adding leading sentences in the sections or ending sections with what we expect in the next section would really help.

0b8508e

> Some restructuring (or adding new sections) could make the content much easier to go through. For example, in the section "How allofplos maintains corpora", you have talked about how to download articles which I think is a use case. Therefore, separating these into different sections will make it more structured.

Downloading PLOS articles is one of the core functions of the package. Made this clearer in 
0b8508e

> The narrative is disconnected: this can be improved by adding trailing sentences after each section hinting at what will be talked about in the next section.

0b8508e

> The writing feels a bit colloquial for a conference proceeding: the authors use "It's geared" as opposed to "It is geared", or "the article data won't" as opposed to "the article data will not" or "If you've used" vs "If you used".

54b1352

> As mentioned by others, the ending is abrupt indeed.

Conclusion in a4fa65f

> " It also quickly and easily parses these article files into Python data structures."
> I would recommend you rewrite this sentence as:
> "It parses these article files into Python data structures efficiently." (not sure that there is any value in "quickly and easily").

54b1352

> "This article will cover how allofplos keeps your articles up-to-date, and how to use it to easily access common article metadata and fuel your meta-research, with actual use cases from inside PLOS."
> I would recommend you break this into two sentenses.

54b1352

> Reference and/or exlapantion of JATS format. (Please assume that the audience for SciPy does not know about JATS format).
> Reference And/Or explanation of DOI. (Please assume that the audience for SciPy does not know about DOI, hows its created/computed, and its importance).

c5be880

> As mentioned by the other reviewers, the ending is not appropriate. The paper needs more organization. To benifit the Scipy audience, it would be better to add more details to your paper. Specifically, by adding some examples of how people have used allofplos.

0b8508e has examples. Conclusion in a4fa65f

> There needs to some mention of related work. Do similar systems exist? Can this be expanded to beyond plos?

6f32d4b discusses expansion beyond PLOS. c5be880 mentions the eLife tool. PMC OA subset in 751df50

> A brief intro plos would be more helpful. Details such as what journals are covered under plos and/or what fields do these journals cover?

0b8508e

eseiver committed:
add PMC OA subset reference

eseiver commented:
@deniederhut do these changes sufficiently address the reviewers' concerns? https://github.com/scipy-conference/scipy_proceedings/pull/417#issuecomment-402944580

deniederhut commented:
That's up to your reviewers! I might ping them to let them know you're
ready for them to take another look.

On Mon, Jul 9, 2018, 13:21 Elizabeth Seiver <notifications@github.com>
wrote:

> @deniederhut <https://github.com/deniederhut> do these changes
> sufficiently address the reviewers' concerns? #417 (comment)
> <https://github.com/scipy-conference/scipy_proceedings/pull/417#issuecomment-402944580>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipy-conference/scipy_proceedings/pull/417#issuecomment-403573725>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGjNrDzl7c4hCXAYZutWvzdRl4FaKsvIks5uE58sgaJpZM4ULpaW>
> .
>

eseiver commented:
Thanks so much for your reviews, @githomin @demba @ankurankan @reddyts. Hopefully you'll find the new version much improved. I've pointed to the corresponding commits that address each point you raised here: https://github.com/scipy-conference/scipy_proceedings/pull/417#issuecomment-402944580

deniederhut commented:
Dear @reddyts @demba @ankurankan @githomin , I know we are all quite preoccupied now that the conference is in full swing, but we would to ask that you take a second look at the paper to see if your concerns have been addressed. Thanks so much!

reddyts commented:
The quality of the paper has improved significantly. Most of the suggested changes have been incorporated. My recommendation at this point is to accept the paper. 

But, I would request the authors to have another look at some of the language in the paper.

One change I would suggest is to change the word `everyone` to `anyone` in the second last sentence of why mine scientific articles section. You can change the sentence to:

Equally importantly because PLOS is Open Access, the opportunity to use this data set is available to anyone capable of downloading and analyzing it.

demba commented:
I agree that the paper has improved significantly from the first version. I would suggest that the authors carefully proofread the paper for typos, as I found a few. The punctuation in the intro could use some work too.

Good job overall.

deniederhut commented:
Sounds like we're really close! Elizabeth, will you have time to run aspell
or something?

On Fri, Jul 13, 2018, 11:26 Demba Ba <notifications@github.com> wrote:

> I agree that the paper has improved significantly from the first version.
> I would suggest that the authors carefully proofread the paper for typos,
> as I found a few. The punctuation in the intro could use some work too.
>
> Good job overall.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipy-conference/scipy_proceedings/pull/417#issuecomment-404883529>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGjNrPWanQ4JhtyfmY-URe2A0eu28Wjtks5uGMoZgaJpZM4ULpaW>
> .
>

eseiver committed:
update article

fix punctuation in intro, fix typos, other wording fixes and tidying up

eseiver commented:
Thanks @demba and @reddyts for your additional feedback. I've fixed the punctuation in the intro, fixed typos throughout, and cleaned up the prose where applicable. dffdf2c
